<!-- -*- mode: XML -*- -->
<!-- run pandar_pointcloud/CloudNodelet in a nodelet manager for a Pandar128 -->

<launch>

  <!-- declare arguments with default values -->
  <arg name="calibration" default="$(find pandar_pointcloud)/params/Pandar128_Correction.csv"/>
  <arg name="device_ip" default="192.168.1.201" />
  <arg name="frame_id" default="PandarSwift" />
  <arg name="manager" default="$(arg frame_id)_nodelet_manager" />
  <arg name="max_range" default="130.0" />
  <arg name="firetime_file" default="$(find pandar_pointcloud)/params/Pandar128_Firetimes.csv"/>
  <arg name="min_range" default="0.5" />
  <arg name="pcap" default="" />
  <arg name="port" default="2368" />
  <arg name="read_fast" default="false" />
  <arg name="read_once" default="false" />
  <arg name="repeat_delay" default="0.0" />
  <arg name="rpm" default="600.0" />
  <arg name="start_angle" default="0.0" />
  <arg name="publish_model" default="both_point_raw" />
  <arg name="namespace" default="hesai"/>
  <arg name="coordinate_correction_flag" default="false" />
  <arg name="channel_config_file" default="$(find pandar_pointcloud)/params/QT128C2X_Channel_Cofig.csv" />
  <!--"cert_file" represents the path of the user's certificate-->
  <arg name="cert_file" default="" />
  <!--"private_key_file" represents the path of the user's private key-->
  <arg name="private_key_file" default="" />
  <!--"ca_file" represents the path of the root certificate-->
  <arg name="ca_file" default="" />

  <!-- start nodelet manager -->
  <node pkg="nodelet" type="nodelet" ns="$(arg namespace)" name="$(arg manager)" args="manager" output="screen" />

  <!-- start cloud nodelet -->
  <include file="$(find pandar_pointcloud)/launch/cloud_nodelet.launch">
    <arg name="calibration" value="$(arg calibration)"/>
    <arg name="manager" value="$(arg manager)" />
    <arg name="max_range" value="$(arg max_range)"/>
    <arg name="min_range" value="$(arg min_range)"/>
    <arg name="start_angle" value="$(arg start_angle)"/>
    <arg name="device_ip" value="$(arg device_ip)" />
    <arg name="frame_id" value="$(arg frame_id)"/>
    <arg name="publish_model" value="$(arg publish_model)"/>
    <arg name="pcap" value="$(arg pcap)"/>
    <arg name="port" value="$(arg port)" />
    <arg name="read_fast" value="$(arg read_fast)"/>
    <arg name="read_once" value="$(arg read_once)"/>
    <arg name="repeat_delay" value="$(arg repeat_delay)"/>
    <arg name="rpm" value="$(arg rpm)"/>
    <arg name="firetime_file" value="$(arg firetime_file)"/>
    <arg name="namespace" value="$(arg namespace)"/>
    <arg name="coordinate_correction_flag" value="$(arg coordinate_correction_flag)"/>
    <arg name="channel_config_file" value="$(arg channel_config_file)"/>
    <arg name="cert_file" value="$(arg cert_file)" />
    <arg name="private_key_file" value="$(arg private_key_file)" />
    <arg name="ca_file" value="$(arg ca_file)" />
  </include>
  <!--
  -->

</launch>
